pkg/builder/local_build_executor.go

```diff
@@ -356,6 +382,13 @@ func (be *localBuildExecutor) Execute(ctx context.Context, filePool re_filesyste
 		response.Result.StderrDigest = stderrDigest.GetPartialDigest()
 	}
 
+	outputParentDirectories, err := newOutputDirectories(command, inputRootDirectory, false)
```

Comment:
This is kind of odd, as it ends up doing `Mkdir()` calls during the upload phase...? I suspect we simply shouldn't share the logic.

Reply:
Note that the third parameter suppresses the Mkdir - everything else is identical

Reply:
Ah, gotcha.